src/Kiota.Builder/Refiners/CliRefiner.cs

```diff
@@ -92,15 +93,25 @@ public override Task Refine(CodeNamespace generatedCode, CancellationToken cance
             AddConstructorsForDefaultValues(generatedCode, false);
             AddSerializationModulesImport(generatedCode);
             RenameDuplicateIndexerNavProperties(generatedCode);
+            // Must be called after the more specific RenameDuplicateIndexerNavProperties
+            RenameMatchingSubsequentNavCommands(generatedCode);
             cancellationToken.ThrowIfCancellationRequested();
             CreateCommandBuilders(generatedCode);
         }, cancellationToken);
     }
     private static void RemoveBackwardCompatibleIndexers(CodeElement currentElement)
-    {//TODO remove for v2
-        if (currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.RequestBuilder) && currentClass.Indexer is CodeIndexer specificIndexer &&
-            !specificIndexer.IndexParameter.Type.Name.Equals("string", StringComparison.OrdinalIgnoreCase) && !(specificIndexer.Deprecation?.IsDeprecated ?? false) &&
-            currentClass.GetChildElements(true).OfType<CodeIndexer>().FirstOrDefault(i => i != specificIndexer && i.IndexParameter.Type.Name.Equals("string", StringComparison.OrdinalIgnoreCase) && (i.Deprecation?.IsDeprecated ?? false)) is CodeIndexer backwardCompatibleIndexer)
+    {
+        //TODO remove for v2
+        if (currentElement is CodeClass
+            {
+                Kind: CodeClassKind.RequestBuilder, Indexer: { } specificIndexer
+            } currentClass &&
+            !specificIndexer.IndexParameter.Type.Name.Equals("string", StringComparison.OrdinalIgnoreCase) &&
+            !(specificIndexer.Deprecation?.IsDeprecated ?? false) &&
+            currentClass.GetChildElements(true).OfType<CodeIndexer>().FirstOrDefault(i =>
+                i != specificIndexer &&
+                i.IndexParameter.Type.Name.Equals("string", StringComparison.OrdinalIgnoreCase) &&
+                (i.Deprecation?.IsDeprecated ?? false)) is { } backwardCompatibleIndexer)
         {
             currentClass.RemoveChildElement(backwardCompatibleIndexer);
         }
```

Comment:
I added a flag since this kind of comparison was leading to side effects during #3158
https://github.com/microsoft/kiota/blob/e0fde4dcaca551c21c52a699c4ffdf5b1bc5b5b1/src/Kiota.Builder/CodeDOM/CodeIndexer.cs#L43

Reply:
So I should use `i.IsLegacyIndexer` instead of `i != specificIndexer && i.IndexParameter.Type.Name.Equals("string", StringComparison.OrdinalIgnoreCase) && (i.Deprecation?.IsDeprecated ?? false)`

Reply:
correct, and you can also update the condition above based of that too